pkg/kubelet/status/state/state_checkpoint_test.go

```diff
@@ -0,0 +1,175 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package state
+
+import (
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/kubernetes/pkg/kubelet/checkpointmanager"
+)
+
+func newTestStateCheckpoint(t *testing.T) *stateCheckpoint {
+	// create temp dir
+	testingDir, err := os.MkdirTemp("", "pod_resource_allocation_state_test")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer func() {
+		if err := os.RemoveAll(testingDir); err != nil {
+			t.Fatal(err)
+		}
+	}()
+	cache := NewStateMemory()
+	checkpointManager, err := checkpointmanager.NewCheckpointManager(testingDir)
+	require.NoError(t, err, "failed to create checkpoint manager")
+	checkpointName := "pod_state_checkpoint"
+	sc := &stateCheckpoint{
+		cache:             cache,
+		checkpointManager: checkpointManager,
+		checkpointName:    checkpointName,
+	}
+	return sc
+}
+
+func Test_stateCheckpoint_storeState(t *testing.T) {
+	sc := newTestStateCheckpoint(t)
+	type args struct {
+		podResourceAllocation PodResourceAllocation
+	}
+
+	tests := []struct {
+		name string
+		args args
+	}{}
+	suffix := []string{"Ki", "Mi", "Gi", "Ti", "Pi", "Ei", "n", "u", "m", "k", "M", "G", "T", "P", "E", ""}
+	factor := []string{"1", "0.1", "0.03", "10", "100", "512", "1000", "1024", "700", "10000"}
+	for _, fact := range factor {
+		for _, suf := range suffix {
+			if (suf == "E" || suf == "Ei") && (fact == "1000" || fact == "10000") {
+				// when fact is 1000 or 10000, suffix "E" or "Ei", the quantity value is overflow
+				// see detail https://github.com/kubernetes/apimachinery/blob/95b78024e3feada7739b40426690b4f287933fd8/pkg/api/resource/quantity.go#L301
+				continue
+			}
+			tests = append(tests, struct {
+				name string
+				args args
+			}{
+				name: fmt.Sprintf("resource - %s%s", fact, suf),
+				args: args{
+					podResourceAllocation: PodResourceAllocation{
+						"pod1": {
+							"container1": {
+								Requests: v1.ResourceList{
+									v1.ResourceCPU:    resource.MustParse(fmt.Sprintf("%s%s", fact, suf)),
+									v1.ResourceMemory: resource.MustParse(fmt.Sprintf("%s%s", fact, suf)),
+								},
+							},
+						},
+					},
+				},
+			})
+		}
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			err := sc.cache.ClearState()
+			require.NoError(t, err, "failed to clear state")
+
+			defer func() {
+				err = sc.checkpointManager.RemoveCheckpoint(sc.checkpointName)
+				require.NoError(t, err, "failed to remove checkpoint")
+			}()
+
+			err = sc.cache.SetPodResourceAllocation(tt.args.podResourceAllocation)
+			require.NoError(t, err, "failed to set pod resource allocation")
+
+			err = sc.storeState()
+			require.NoError(t, err, "failed to store state")
```

Comment:
Wouldn't it be better to just call `sc.SetPodResourceAllocation(...)` directly? I think it does this internally.

Reply:
Yeah, I'll update it